src/rust/engine/dep_inference/src/python/mod.rs

```diff
@@ -322,7 +322,9 @@ impl Visitor for ImportCollector<'_> {
     fn visit_string(&mut self, node: tree_sitter::Node) -> ChildBehavior {
         let range = node.range();
         let text: &str = self.string_at(range);
-        if !text.contains(|c: char| c.is_ascii_whitespace() || c == '\\') {
+        if !text.contains(|c: char| c.is_ascii_whitespace() || c == '\\')
+            && !self.is_pragma_ignored(node.parent().unwrap())
+        {
```

Comment:
no ignore for ends with period check in the rust version?

Reply:
This string thing captures strings for both imports and assets. A module can't end in a `.`, but an asset can.